packages/driver/cypress/integration/e2e/multi-domain/commands/multi_domain_navigation.spec.ts

```diff
@@ -1,43 +1,243 @@
-  // Your page did not fire its `load` event within `undefinedms`.
-  // You can try increasing the `pageLoadTimeout` value in `undefined` to wait longer.
-  // Browsers will not fire the `load` event until all stylesheets and scripts are done downloading.
-  // When this `load` event occurs, Cypress will continue running commands.
-  // at timedOutWaitingForPageLoad (webpack:///../driver/src/cy/commands/navigation.ts?:59:72)
-  // at eval (webpack:///../driver/src/cy/commands/navigation.ts?:1143:16)
-  it.skip('.visit()', () => {
+  context('.visit()', () => {
+    it('calls the correct load handlers', () => {
+      const primaryCyBeforeLoadSpy = cy.spy()
+      const primaryCyLoadSpy = cy.spy()
+      const primaryVisitBeforeLoadSpy = cy.spy()
+      const primaryVisitLoadSpy = cy.spy()
+
+      cy.on('window:before:load', primaryCyBeforeLoadSpy)
+      cy.on('window:load', primaryCyLoadSpy)
+
+      cy.visit('/fixtures/multi-domain.html', {
+        onBeforeLoad: primaryVisitBeforeLoadSpy,
+        onLoad: primaryVisitLoadSpy,
+      }).then(() => {
+        expect(primaryCyBeforeLoadSpy).to.be.calledOnce
+        expect(primaryCyLoadSpy).to.be.calledTwice // twice because it's also called for 'about:blank'
+        expect(primaryVisitBeforeLoadSpy).to.be.calledOnce
+        expect(primaryVisitLoadSpy).to.be.calledOnce
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        const secondaryCyBeforeLoadSpy = cy.spy()
+        const secondaryCyLoadSpy = cy.spy()
+        const secondaryVisitBeforeLoadSpy = cy.spy()
+        const secondaryVisitLoadSpy = cy.spy()
+
+        cy.on('window:before:load', secondaryCyBeforeLoadSpy)
+        cy.on('window:load', secondaryCyLoadSpy)
+
+        cy.visit('http://www.foobar.com:3500/fixtures/dom.html', {
+          onBeforeLoad: secondaryVisitBeforeLoadSpy,
+          onLoad: secondaryVisitLoadSpy,
+        }).then(() => {
+          expect(secondaryCyBeforeLoadSpy).to.be.calledOnce
+          expect(secondaryCyLoadSpy).to.be.calledOnce
+          expect(secondaryVisitBeforeLoadSpy).to.be.calledOnce
+          expect(secondaryVisitLoadSpy).to.be.calledOnce
+        })
+      }).then(() => {
+        expect(primaryCyBeforeLoadSpy).to.be.calledOnce
+        expect(primaryCyLoadSpy).to.be.calledTwice
+        expect(primaryVisitBeforeLoadSpy).to.be.calledOnce
+        expect(primaryVisitLoadSpy).to.be.calledOnce
+      })
+    })
+
+    it('supports visiting primary first', () => {
+      cy.visit('/fixtures/multi-domain.html')
+
+      cy.switchToDomain('foobar.com', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/multi-domain-secondary.html')
+
+        cy.get('[data-cy="dom-check"]').should('have.text', 'From a secondary domain')
+      })
+    })
+
+    it('supports skipping visiting primary first', () => {
+      cy.switchToDomain('foobar.com', () => {
+        cy.visit('http://www.foobar.com:3500/fixtures/multi-domain-secondary.html')
```

Comment:
Yes, in this case the primary is what the `baseUrl` is set to (`localhost:3500`). If this `cy.visit` was moved outside of the `switchToDomain` it would change top to `foobar.com` and the primary would not be `foobar.com`.